gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ReplicationUtils.java

```
@@ -0,0 +1,112 @@
+    Preconditions.checkArgument(config.hasPath(REPLICATION_REPLICAS), "missing required config entery " + REPLICATION_REPLICAS);
+
+    Config replicaConfig = config.getConfig(REPLICATION_REPLICAS);
+    Preconditions.checkArgument(replicaConfig.hasPath(REPLICATOIN_REPLICAS_LIST), "missing required config entery " + REPLICATOIN_REPLICAS_LIST);
+    List<String> replicas = replicaConfig.getStringList(REPLICATOIN_REPLICAS_LIST);
+
+    List<ReplicationReplica> result = new ArrayList<ReplicationReplica>();
+    for (String replicaName : replicas) {
+      Config subConfig = replicaConfig.getConfig(replicaName);
+      result.add(
+          new ReplicationReplica(replicaName, ReplicationUtils.buildReplicationLocation(subConfig), 
+              ReplicationUtils.buildDatasetFinder(subConfig)));
+    }
+
+    return result;
+  }
+
+  public static DataFlowTopology buildDataFlowTopology(Config config, ReplicationSource source,
+      List<ReplicationReplica> replicas) {
+    Preconditions.checkArgument(config.hasPath(DATAFLOWTOPOLOGY), "missing required config entery " + DATAFLOWTOPOLOGY);
+    Config dataflowConfig = config.getConfig(DATAFLOWTOPOLOGY);
+
+    return new DataFlowTopology.Builder().withReplicationSource(source).withReplicationReplicas(replicas)
+        .withTopologyConfig(dataflowConfig).build();
+  }
+
+  public static ReplicationLocation buildReplicationLocation(Config config) {
+    Preconditions.checkArgument(config.hasPath(REPLICATION_LOCATION_TYPE_KEY), "missing required config entery " + REPLICATION_LOCATION_TYPE_KEY);
+
+    String type = config.getString(REPLICATION_LOCATION_TYPE_KEY);
+    Preconditions.checkArgument(config.hasPath(type));
+
+    ReplicationType RType = ReplicationType.forName(type);
+
+    if (RType == ReplicationType.HDFS) {
+      return new HdfsReplicationLocation(config.getConfig(type));
+    }
+
+    // TODO, other replication types implmenetation 
+    return null;
```

Can you instead throw an `UnsupportedOperationException` with the explanation?